composer/lib/dependabot/composer/metadata_finder.rb

```diff
@@ -28,27 +28,26 @@ def look_up_source_from_packagist
         return nil if packagist_listing&.fetch("packages", nil) == []
         return nil unless packagist_listing&.dig("packages", dependency.name.downcase)
 
-        version_listings =
-          packagist_listing["packages"][dependency.name.downcase].
-          select { |version, _| Composer::Version.correct?(version) }.
-          sort_by { |version, _| Composer::Version.new(version) }.
-          map { |_, listing| listing }.
-          reverse
-
-        potential_source_urls =
-          version_listings.
-          flat_map { |info| [info["homepage"], info.dig("source", "url")] }.
-          compact
-
-        source_url = potential_source_urls.find { |url| Source.from_url(url) }
```

Comment:
When I initially started working on this, I was afraid I'd have to navigate to the exact version of the dependency that we wanted, then extract the source URL from there. That'd probably be a more accurate way to handle it. But it'd be painful in minified code since we'd need to un-minify it to determine the correct source link that matched to that particular version.

But the previous code kept it simple, and merely iterated on the list of URLs from newest to oldest until it found the first viable candidate and then used that, regardless of whether it matched the version. So by continuing to keep things simple with the same behavior, we avoid any need to un-minify.

Reply:
Un-minify here would really only mean while you iterate through versions you keep the last good source_url you found, once you find the correct version field you return the last good url and done. But it also seems reasonable as is because in 99% of cases URLs will be the same for all versions